src/app/projects/services/geo.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable } from '@angular/core';
+import {HttpClient} from "@angular/common/http";
+import {IngestService} from "@shared/services/ingest.service";
+import {Observable, Subject} from "rxjs";
+import {Project} from "@shared/models/project";
+import {catchError} from "rxjs/operators";
+import {LoaderService} from "@shared/services/loader.service";
+import {BrokerService} from "@shared/services/broker.service";
+import {AlertService} from "@shared/services/alert.service";
+import {SaveFileService} from "@shared/services/save-file.service";
+import {Router} from "@angular/router";
+
+@Injectable()
+export class GeoService {
+  loading: Subject<boolean> = new Subject<boolean>();
+
+  constructor(private http: HttpClient,
+              private ingestService: IngestService,
+              private loaderService: LoaderService,
+              private brokerService: BrokerService,
+              private alertService: AlertService,
+              private saveFileService: SaveFileService,
+              private router: Router) {
+  }
+
+  importProjectUsingGeo(geoAccession) {
+    this.loading.next(true)
+    this.getProjectsWithGeo(geoAccession)
+      .subscribe(projects => {
+          this.showExistingProjectsAlert(projects, 'geo accession');
+          if (projects.length == 0) {
+            this.onUniqueGeoAccession(geoAccession);
+          }
+          this.loading.next(false)
+        },
+        error => {
+          this.alertService.error('An error occurred', error.message);
+          this.loading.next(false);
+        }
+      );
+  }
+
+  private getProjectsWithGeo(geoAccession: string): Observable<Project[]> {
+    const criteria = {
+      'field': 'content.geo_series_accessions',
+      'operator': 'IN',
+      'value': [geoAccession]
+    };
+    return this.ingestService.getProjectsUsingCriteria(criteria);
+  }
+
+  private onUniqueGeoAccession(geoAccession) {
+    this.loaderService.display(true, 'This may take a moment. Please wait...');
+    this.brokerService.importProjectUsingGeo(geoAccession).pipe(
+      catchError(err => {
+        this.loaderService.display(true, `Sorry, we are unable to import the project yet due to error: [${err.message}]. You can still get a spreadsheet to import the project later.
```

Comment:
`Sorry, we are unable...` -> `Unable to...`

Reply:
Also remove "yet"